src/main/java/net/sf/jabref/logic/integrity/NoBibtexFieldChecker.java

```diff
@@ -1,34 +1,43 @@
 package net.sf.jabref.logic.integrity;
 
+import java.util.Arrays;
 import java.util.List;
-import java.util.SortedSet;
-import java.util.TreeSet;
+import java.util.Set;
 import java.util.stream.Collectors;
 
 import net.sf.jabref.logic.integrity.IntegrityCheck.Checker;
 import net.sf.jabref.logic.l10n.Localization;
 import net.sf.jabref.model.entry.BibEntry;
-import net.sf.jabref.model.entry.EntryConverter;
+import net.sf.jabref.model.entry.BibLatexEntryTypes;
+import net.sf.jabref.model.entry.BibtexEntryTypes;
 import net.sf.jabref.model.entry.FieldName;
 
 /**
- * This checker checks whether the entry does not contain any field appearing only in BibLaTeX (and not in BibTex)
+ * This checker checks whether the entry does not contain any field appearing only in BibLaTeX (and not in BibTeX)
  */
 public class NoBibtexFieldChecker implements Checker {
 
-    @Override
-    public List<IntegrityMessage> check(BibEntry entry) {
-        SortedSet<String> allBibLaTeXOnlyFields = new TreeSet<>();
-        allBibLaTeXOnlyFields.addAll(EntryConverter.FIELD_ALIASES_LTX_TO_TEX.keySet());
 
-        // file is both in BibTeX and BibLaTeX
-        allBibLaTeXOnlyFields.remove(FieldName.FILE);
+    private List<String> getAllBiblatexOnlyFields() {
+        Set<String> allBibtexFields = BibtexEntryTypes.ALL.stream().flatMap(type -> type.getAllFields().stream()).collect(Collectors.toSet());
+
+        // fields as set at net.sf.jabref.preferences.JabRefPreferences.setLanguageDependentDefaultValues()
+        final List<String> JABREF_GENERAL_FIELDS =  Arrays.asList(FieldName.CROSSREF, FieldName.KEYWORDS, FieldName.FILE, FieldName.DOI, FieldName.URL, FieldName.COMMENT, FieldName.OWNER, FieldName.TIMESTAMP, FieldName.ABSTRACT, FieldName.REVIEW);
```

Comment:
The naming would indicate that it is a static variable, but it is not. Same applies to `ALL_BIBLATEX...` below.
Moreover, if this list is reused somewhere else then please extract it and reuse it (e.g. to FieldName ?)